### PR TITLE
build(bazel): add `data` attr to ng_package

### PR DESCRIPTION
**Pull request by jelbourn**
@alexeagle One weird thing now is that you can put static files (css, markdown, scss, etc.) into either `srcs` or `data` and they will work the same way. I'm not a fan of having two valid ways to do this, which goes back to the question of whether `srcs` should only accept `ts_library` and `ng_module`.

I wasn't sure how to add a test for the error case when `srcs` includes the output of another rule. Also, it's just checking `binDir` now, but should we also be doing stuff for `bazel-genfiles`?

### CODE REVIEW

File: packages/bazel/test/ng_package/example/BUILD.bazel [hunk 1]
```diff
@@ -15,9 +15,22 @@ ng_package(
...
     ],
 )
+
+# Use a genrule to create a file in bazel-bin to ensure that the bin output of
+# a rule can be passed through to the `data` of ng_package.
+genrule(
+    name = "arbitrary_text_file",
+    outs = ["arbitrary_text.txt"],
+    cmd = "echo Hello > $@",
+    output_to_bindir = True,
```
Comment (alexeagle) [inline packages/bazel/test/ng_package/example/BUILD.bazel]: also add one with `output_to_bindir = False` to test the bazel-genfiles prefix gets stripped?

File: packages/bazel/src/ng_package/ng_package.bzl [hunk 2]
```diff
@@ -212,6 +213,7 @@ def _ng_package_impl(ctx):
   packager_args.add(_flatten_paths(esm5), join_with=",")
   packager_args.add(_flatten_paths(bundles), join_with=",")
   packager_args.add([s.path for s in ctx.files.srcs], join_with=",")
+  packager_args.add([d.path for d in ctx.files.data], join_with=",")
```
Comment (alexeagle) [inline packages/bazel/src/ng_package/ng_package.bzl]: I don't think this is right - ctx.files.data will only receive the default outputs of the targets listed in data. for example, if you put a `ts_library` in the `data` (a strange thing to do) I think you'll find that the `.d.ts` files go in the bundle but not the `.js` files.

File: packages/bazel/src/ng_package/packager.ts [hunk 3]
```diff
@@ -77,38 +83,38 @@ function main(args: string[]): number {
...
+   * @param inputPath Path to the file in the input tree.
+   * @param fileContent Content of the file.
    */
-  function writeSrcFile(f: string, c: string) {
-    mkDirWriteFile(path.join(out, path.relative(srcDir, f)), c);
+  function writeFileFromInputPath(inputPath: string, fileContent: string) {
+    // We want the relative path from the given file to its ancestor "root" directory.
+    // This root depends on whether the file lives in the source tree (srcDir) as a basic file
+    // input to ng_package, or in the out tree (binDir) as the output of another rule.
+    const rootDir = inputPath.includes(binDir) ? binDir : srcDir;
```
Comment (alexeagle) [inline packages/bazel/src/ng_package/packager.ts]: should check genfilesDir too

File: packages/bazel/src/ng_package/ng_package.bzl (hunk 2 above)
Comment (jelbourn) [reply packages/bazel/src/ng_package/ng_package.bzl]: Should it be all outputs?

File: packages/bazel/test/ng_package/example/BUILD.bazel (hunk 1 above)
Comment (jelbourn) [reply packages/bazel/test/ng_package/example/BUILD.bazel]: Done

File: packages/bazel/src/ng_package/packager.ts [hunk 4]
```diff
@@ -14,6 +14,9 @@ function main(args: string[]): number {
   // Exit immediately when encountering an error.
   shx.set('-e');
 
+  // Keep track of whether an error has occured so that we can return an appropriate exist code.
```
Comment (alfaproject) [inline packages/bazel/src/ng_package/packager.ts]: exit

**Review by alexeagle — APPROVED** [review]

File: packages/bazel/src/ng_package/ng_package.bzl (hunk 2 above)
Comment (jelbourn) [reply packages/bazel/src/ng_package/ng_package.bzl]: Done

**Comment by jelbourn** [comment]
For caretaker: this PR doesn't affect g3

**Pull request closed without merging**